src/main/java/build/buildfarm/common/grpc/TracingMetadataUtils.java

```diff
@@ -35,6 +37,8 @@ private TracingMetadataUtils() {}
   private static final Context.Key<RequestMetadata> CONTEXT_KEY =
       Context.key("remote-grpc-metadata");
 
+  private static ImmutableList<Property> capturedHeaders = ImmutableList.of();
+
```

Comment:
This should not be static to TracingMetadataUtils - it does not follow by guarantee the current context, nor is it threadsafe or aware.